src/mainActions.ts

```diff
@@ -1,144 +1,169 @@
-import * as electronOs from "os"
-import { ipcMain, dialog, BrowserWindow } from 'electron';
+import * as electronOs from "os";
+import { ipcMain, dialog, BrowserWindow } from "electron";
 
 import {
   ErrorInfo,
   OPEN_DIALOG_CHANNEL,
   OPEN_ERROR_DIALOG_CHANNEL,
   IMPORT_REPLY_CHANNEL,
-  SAVE_REPLY_CHANNEL, 
+  SAVE_REPLY_CHANNEL,
   JSON_IMPORT_REPLY_FORMAT,
   SQLITE_IMPORT_REPLY_FORMAT,
   SQLITE_SAVE_REPLY_FORMAT,
   SQLITE_PATH_FOR_JSON_REQUEST_FORMAT,
   UTIL_CHANNEL,
-} from './constants/IpcConnection';
+} from "./constants/IpcConnection";
 
-export function chooseSqlitePathForJsonImport(mainWindow: BrowserWindow, jsonFilePath: string){
+export function chooseSqlitePathForJsonImport(
+  mainWindow: BrowserWindow,
+  jsonFilePath: string
+) {
   console.log("Prompt to choose Sqlite path for JSON file");
 
   let defaultPath;
-  if (electronOs.platform() === 'linux') {
+  if (electronOs.platform() === "linux") {
     // remove the .json (or other) extention of the path.
-    defaultPath = jsonFilePath.substring(0, jsonFilePath.lastIndexOf('.')) + '.sqlite';
+    defaultPath =
+      jsonFilePath.substring(0, jsonFilePath.lastIndexOf(".")) + ".sqlite";
   } else {
     // FIXME: this is some ugly regex used to get filename with no extension.
     // see: https://stackoverflow.com/questions/4250364/how-to-trim-a-file-extension-from-a-string-in-javascript
```

Comment:
We need to document the file license when we copy code from elsewhere, and also have a common notice for all files like we have in scancode-toolkit and elsewhere: for example see https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/cache.py#L4

Reply:
This one specifically under https://creativecommons.org/licenses/by-sa/3.0/ yes. We should probably put a line and reference to the text, but I'm not sure of the effect of a sharealike license on the whole file/repo license. @pombredanne @DennisClark what would be recommended here?

Reply:
These regex are too small to be relevant copyright-wise IMHO... but this code is also arcane, and this is best rewritten with a cleaner and more readable way.